boxoffice/views/item_collection.py

```diff
@@ -1,35 +1,43 @@
 # -*- coding: utf-8 -*-
 
-from flask import make_response, render_template, jsonify, request, Markup
+import json
+from flask import render_template, jsonify, request, Markup
 from pycountry import pycountry
 from coaster.views import load_models
 from coaster.utils import getbool
 from boxoffice import app
-from boxoffice.models import Organization, ItemCollection, Item
-from utils import xhr_only, cors
+from boxoffice.models import Organization, ItemCollection, Item, DiscountPolicy
+from utils import xhr_only, cors, sanitize_coupons
 from boxoffice.data import indian_states
 
 
 def jsonify_item(item):
+    if item.restricted_entry:
+        code_list = request.args.get('code') and sanitize_coupons(json.loads(request.args.get('code')))
```

Comment:
Why are we accepting JSON in a GET parameter? If you need to accept a list, use `request.args.getlist('code')`

Reply:
Changed to use `request.args.getlist('code')`.